qcodes/instrument/parameter.py

```diff
@@ -1545,3 +1547,197 @@ def __init__(self, name, instrument=None, initial_value=None, **kwargs):
+    value, and store the gain/division in the metadata.
+
+    Examples:
+        Resistive voltage divider
+        >>> vd = ParameterScaler(dac.chan0, division = 10)
+
+        Voltage multiplier
+        >>> vb = ParameterScaler(dac.chan0, gain = 30, name = 'Vb')
+
+        Transimpedance amplifier
+        >>> Id = ParameterScaler(multimeter.amplitude, division = 1e6, name = 'Id', unit = 'A')
+
+    Args:
+        output: Physical Parameter that need conversion
+        division: the division value
+        gain: the gain value
+        label: label of this parameter, by default uses 'output' label
+            but attaches _amplified or _attenuated depending if gain
+            or division has been specified
+        name: name of this parameter, by default uses 'output' name
+            but attaches _amplified or _attenuated depending if gain
+            or division has been specified
+        unit: resulting unit. It uses the one of 'output' by default
+    """
+
+    class Role(enum.Enum):
+        GAIN = enum.auto()
+        DIVISION = enum.auto()
+
+
+    def __init__(self,
+                 output: Parameter,
+                 division: Union[int, float, Parameter] = None,
+                 gain: Union[int, float, Parameter] = None,
+                 name: str=None,
+                 label: str=None,
+                 unit: str=None) -> None:
+        # Set the name
+        if name:
+            self.name = name
+        else:
+            self.name = "{}_scaled".format(output.name)
+
+        # Set label
+        if label:
+            self.label = label
+        elif name:
+            self.label = name
+        else:
+            self.label = "{}_scaled".format(output.label)
+
+        # Set the unit
+        if unit:
+            self.unit = unit
+        else:
+            self.unit = output.unit
+
+        super().__init__(
+            name=self.name,
+            label=self.label,
+            unit=self.unit
+            )
+
+        self._wrapped_parameter = output
+        self._wrapped_instrument = getattr(output, "_instrument", None)
+
+        # Set the role, either as divider or amplifier
+        # Raise an error if nothing is specified
+        is_divider = division is not None
+        is_amplifier = gain is not None
+
+        if not xor(is_divider, is_amplifier):
+            raise ValueError('Provide only division OR gain')
+
+        if is_divider:
+            self.role = ParameterScaler.Role.DIVISION
+            self._multiplier = division
+        elif is_amplifier:
+            self.role = ParameterScaler.Role.GAIN
+            self._multiplier = gain
+
+        # extend metadata
+        self._meta_attrs.extend(["division"])
+        self._meta_attrs.extend(["gain"])
+        self._meta_attrs.extend(["role"])
+        self.metadata['wrapped_parameter'] = self._wrapped_parameter.name
+        try:
+            self.metadata['wrapped_instrument'] = self._wrapped_instrument.name
```

Comment:
instead of try-catch please use `getattr` function with an "if" statement:
```python
if self._wrapped_instrument:
    self.metadata['wrapped_instrument'] = getattr(self._wrapped_instrument, "name", None)
```

Reply:
ok